test/Tests/src/Semantic/Conversion_Spec.enso

```diff
@@ -65,6 +65,16 @@ type Fool
 
 Fool.from (that : Any) = Fool.Value that
 
+type Blob
+    Text c:Text
+    Binary b:File
+    Json anything
+
+
+Blob.from (that:Text) = Blob.Text that
```

Comment:
This line is a test for 
- #7770

without the fix this conversion isn't found.

Reply:
Oh, okay.

That is not at all clear - I didn't know that `Any.to` wouldn't find it without it.

---

I was expecting a test where we have a _type_ and its _conversion_ defined in one file, and usage in another file where only the type but not the conversion is used - that seems to be the more typical and expected usage of this fix (although ofc I appreciate the fix for `Any.to` was needed as well).

Can we have a test for this case as well?

Essentially, we'd have to move one type from these `Conversion_Spec` together with its conversion, to a separate file. Then we can import the type and check that it still finds it.

Reply:
The mentioned test case is called "Avoid Any.to conversion of Blob into Blob", so it primarily tests _avoiding_ the conversion. I wouldn't know that if it failed, it may be related to #7770.

It would be good to have a separate test case about it - `Test.specify "search the target type scope for conversions"`.

Reply:
Another interesting question, what happens if we have conflicting possible conversions??

i.e. I have file `A.enso`:
```
type My_A
    Inner x
	Outer x

My_A.from (that:Text) = My_A.Inner that
```
and then a file `Main.enso`:
```
import project.A.My_A

My_A.from (that:Text) = My_A.Outer that

foo (m:My_A) = m.to_text

main =
    IO.println (foo "TEST")
```
What will happen?

I expect we should get some _ambiguous conversion_ failure at runtime, because it is impossible to determine which conversion should be used. Or will the compiler detect the redundancy at compile-time, even though the conversions "do not see each other"?

Reply:
Related bug: https://github.com/enso-org/enso/issues/7853

Reply:
> I expect we should get some ambiguous conversion failure at runtime

Of course we are not going to get any _ambiguous failure_! Enso is a dynamic language and thus it will choose some. The [current order can inspected here](https://github.com/enso-org/enso/pull/7849/files#diff-c3869665e0ec34703f8e3eea0464924c0e4260eab392dbccdc202ec1f0270379R207).

Reply:
Looking at the code, if I define a conversion in my own file, whenever a conversion defined next to the type definition exists, my conversion will _never_ be called. Right?

Then, as a user, when I define my conversion - I guess it will be confusing that my conversion is not called even if it's defined in the same file as the usage. Shouldn't I get an error (if not at compile time since that is not possible, then at run time)? 

How else can I learn about the conversion resolution?

Reply:
Yes, I wasn't happy about the order of resolution either.  I'd say _own conversion_ or _imported conversions_ shall win over the one at _original class_. It's kinda convoluted.